src/main/kotlin/com/cognifide/gradle/aem/deploy/tasks/AbstractTask.kt

```diff
@@ -169,4 +176,79 @@ abstract class AbstractTask : DefaultTask(), AemTask {
         return response
     }
 
+    protected fun deletePackage(installedPackagePath: String, sync: DeploySynchronizer) {
+        val url = sync.htmlTargetUrl + installedPackagePath + "/?cmd=delete"
+
+        logger.info("Deleting package using command: " + url)
+
+        try {
+            val rawHtml = sync.post(url)
+            val response = DeleteResponse(rawHtml)
+
+            when (response.status) {
+                AbstractHtmlResponse.Status.SUCCESS,
+                AbstractHtmlResponse.Status.SUCCESS_WITH_ERRORS -> if (response.errors.isEmpty()) {
+                    logger.info("Package successfully deleted.")
+                } else {
+                    logger.warn("Package deleted with errors.")
+                    response.errors.forEach { logger.error(it) }
+                    throw DeployException("Package deleted with errors!")
+                }
+                AbstractHtmlResponse.Status.FAIL -> {
+                    logger.error("Package deleting failed.")
+                    response.errors.forEach { logger.error(it) }
+                    throw DeployException("Package deleting failed!")
+                }
+            }
+
+        } catch (e: Exception) {
+            throw DeployException("Cannot delete package.", e)
+        }
+    }
+
+    protected fun uninstallPackage(installedPackagePath: String, sync: DeploySynchronizer) {
+        val url = sync.htmlTargetUrl + installedPackagePath + "/?cmd=uninstall"
+
+        logger.info("Uninstalling package using command: " + url)
+
+        try {
+            val rawHtml = sync.post(url, mapOf(
+                    "recursive" to config.recursiveInstall,
+                    "acHandling" to config.acHandling
+            ))
+            val response = UninstallResponse(rawHtml)
+
+            when (response.status) {
+                AbstractHtmlResponse.Status.SUCCESS,
+                AbstractHtmlResponse.Status.SUCCESS_WITH_ERRORS -> if (response.errors.isEmpty()) {
+                    logger.info("Package successfully uninstalled.")
+                } else {
+                    logger.warn("Package uninstalled with errors.")
+                    response.errors.forEach { logger.error(it) }
+                    throw DeployException("Package uninstalled with errors!")
+                }
+                AbstractHtmlResponse.Status.FAIL -> {
+                    logger.error("Package uninstalling failed.")
+                    response.errors.forEach { logger.error(it) }
+                    throw DeployException("Package uninstalling failed!")
+                }
+            }
+
+        } catch (e: Exception) {
+            throw DeployException("Cannot uninstall package.", e)
+        }
+    }
+
+    protected fun ensureUserAwareness(taskName: String) {
+        if (project.properties.containsKey(USER_AWARENESS_PROP)) return
+
+        throw DeployException(
+                "**********\n" +
```

Comment:
You introduced special notation for warning. If you want to continue it, please wrap into some helper, because what if while developing another feature someone would like to format it in the same way? another hardcoded string format which will potentially look a little bit different, because of mislooking.

Generally, I prefer to do not use special formatting for this message and keep it simple.

`throw DeployException("Warning! This task execution must be confirmed by specyfing explicitly parameter '-P$FORCE_PROP'. It is recommended to firstly protect against potential data loss by checking out content using '${SyncTask.NAME}' task.  Continue only if risk is identified.")